packages/desktop-client/src/components/reports/CategorySpending.js

```diff
@@ -0,0 +1,161 @@
+          ),
+      ),
+    );
+  }, [start, end, numberOfMonthsAverage, categories, selectedCategories]);
+  const perCategorySpending = useReport('category_spending', getGraphData);
+
+  useEffect(() => {
+    getCategories().then(categories => {
+      setCategories(categories);
+      setSelectedCategories(categories.list);
+    });
+  }, []);
+
+  useEffect(() => {
+    async function run() {
+      const trans = await send('get-earliest-transaction');
+      const currentMonth = monthUtils.currentMonth();
+      let earliestMonth = trans
+        ? monthUtils.monthFromDate(d.parseISO(fromDateRepr(trans.date)))
+        : currentMonth;
+
+      // Make sure the month selects are at least populates with a
+      // year's worth of months. We can undo this when we have fancier
+      // date selects.
+      const yearAgo = monthUtils.subMonths(monthUtils.currentMonth(), 12);
+      if (earliestMonth > yearAgo) {
+        earliestMonth = yearAgo;
+      }
+
+      const allMonths = monthUtils
+        .rangeInclusive(earliestMonth, monthUtils.currentMonth())
+        .map(month => ({
+          name: month,
+          pretty: monthUtils.format(month, 'MMMM, yyyy'),
+        }))
+        .reverse();
+
+      setAllMonths(allMonths);
+    }
+    run();
+  }, []);
+
+  function onChangeDates(start, end) {
+    setStart(start);
+    setEnd(end);
+  }
+
+  if (!allMonths || !perCategorySpending) {
+    return null;
+  }
+
+  const numberOfMonthsOptions = [
+    { value: 1, description: 'No averaging' },
+    { value: 3, description: '3 months' },
+    { value: 6, description: '6 months' },
+    { value: 12, description: '12 months' },
+    { value: -1, description: 'All time' },
+  ];
+  const numberOfMonthsLine = numberOfMonthsOptions.length - 1;
+
+  return (
+    <View style={[styles.page, { overflow: 'hidden' }]}>
+      <Header
+        title="Category Spending"
+        allMonths={allMonths}
+        start={start}
+        end={end}
+        onChangeDates={onChangeDates}
+        numberOfMonths={numberOfMonthsAverage}
+        numberOfMonthsOptions={numberOfMonthsOptions}
+        numberOfMonthsLine={numberOfMonthsLine}
+        onChangeNumberOfMonths={setNumberOfMonthsAverage}
+        categorySelectorVisible={categorySelectorVisible}
+        onChangeCategoryVisible={setCategorySelectorVisible}
+      />
+      <View
+        style={{ display: 'flex', flexDirection: 'row', padding: 15, gap: 15 }}
+      >
+        <View
+          style={{
+            height: '360',
+            overflowY: 'scroll',
+            width: !categorySelectorVisible ? 0 : 'auto',
+          }}
+        >
+          <CategorySelector
+            categoryGroups={categories.grouped.filter(
+              categoryGroup => !categoryGroup.is_income,
+            )}
+            selectedCategories={selectedCategories}
+            setSelectedCategories={setSelectedCategories}
+            onChange={e => console.log(e)}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏we shouldn't leave unnecessary console logs in the code.
```suggestion
```